tgui/packages/tgui/interfaces/AirAlarm.tsx

```diff
@@ -0,0 +1,434 @@
+import { Fragment } from 'inferno';
+import { useBackend, useLocalState } from '../backend';
+import {
+  Box,
+  Button,
+  LabeledList,
+  Modal,
+  NumberInput,
+  Section,
+  Table,
+} from '../components';
+import { Window } from '../layouts';
+import { Scrubber, Vent, VentData, ScrubberData } from './common/AtmosControls';
+import { InterfaceLockNoticeBox } from './common/InterfaceLockNoticeBox';
+
+type AlarmThreshold = {
+  warning_min: number;
+  hazard_min: number;
+  warning_max: number;
+  hazard_max: number;
+  threshold_name: string;
+  threshold_unit: string;
+};
+type GasInformation = { name: string; value: string; status: 0 | 1 | 2 };
+type AirAlarmStaticData = {
+  modes: {
+    name: string;
+    mode: number;
+    danger: boolean;
+  }[];
+  warning_min: number;
+  warning_max: number;
+  hazard_min: number;
+  hazard_max: number;
+};
+type AirAlarmDynamicData = {
+  locked: boolean;
+  siliconUser: boolean;
+  emagged: boolean;
+  danger_level: number;
+  atmos_alarm: boolean;
+  fire_alarm: boolean;
+  sensor_reading: { [criteria: string]: GasInformation };
```

Comment:
You can use `Record<string, GasInformation>` here for less brackets and more semantics.